File: src/test/java/com/snackgame/server/member/StatusServiceTest.java
package com.snackgame.server.member;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import com.snackgame.server.member.domain.Member;
import com.snackgame.server.member.domain.StatusService;
import com.snackgame.server.support.general.ServiceTest;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@ServiceTest
class StatusServiceTest {

    @Autowired
    private StatusService statusService;

    @Autowired
    private MemberService memberService;

    @Test
    void 레벨업을_테스트한다() {
        //given
        Member member = memberService.createWith("박정환");

        //when
        statusService.updateStatus(member.getStatus(), 1080);

        //then
        Assertions.assertThat(member.getStatus().getLevel()).isEqualTo(4);
    }

}

File: src/test/java/com/snackgame/server/status/StatusServiceTest.kt
@file:Suppress("NonAsciiCharacters")

package com.snackgame.server.status

import com.snackgame.server.game.session.event.SessionEndEvent
import com.snackgame.server.history.fixture.SnackgameFixture
import com.snackgame.server.member.fixture.MemberFixture
import com.snackgame.server.member.fixture.MemberFixture.땡칠
import com.snackgame.server.member.service.MemberAccountService
import com.snackgame.server.support.general.ServiceTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

@ServiceTest
class StatusServiceTest {

    @Autowired
    private lateinit var statusService: StatusService

    @Autowired
    private lateinit var memberAccountService: MemberAccountService

    @BeforeEach
    fun setUp() {
        MemberFixture.saveAll()
    }

    @Test
    fun `게임이 끝날때 게임 점수만큼 경험치를 부여한다`() {
        val session = SnackgameFixture.first()
        statusService.onSessionEnd(SessionEndEvent.of(session))

        val member = memberAccountService.getBy(땡칠().id)
        assertThat(member.status.exp).isEqualTo(session.score.toDouble())
    }
}
